cmd/nerdctl/container_run_network_linux_test.go

```diff
@@ -302,6 +302,46 @@ func TestUniqueHostPortAssignement(t *testing.T) {
 	}
 }
 
+func TestHostPortAlreadyInUse(t *testing.T) {
+	testCases := []struct {
+		hostPort      string
+		containerPort string
```

Comment:
Adding `protocol` tests helps to improve test coverage.

Reply:
Added 2 more tests for protocol.